e2e/tests-dfx/create.bash

```diff
@@ -83,33 +102,33 @@ teardown() {
     assert_match "dns error: failed to lookup address information"
 }
 
-
 @test "create accepts --controller <controller> named parameter, with controller by identity name" {
     dfx_start
     dfx identity new alice
     ALICE_PRINCIPAL=$(dfx --identity alice identity get-principal)
+    ALICE_WALLET=$(dfx --identity alice identity get-wallet)
 
-    assert_command dfx canister create --all --controller alice
+    assert_command dfx canister create --all --controller "${ALICE_PRINCIPAL}"
     assert_command dfx canister info e2e_project
+    assert_not_match "Controllers: ($ALICE_WALLET $ALICE_PRINCIPAL|$ALICE_PRINCIPAL $ALICE_WALLET)"
     assert_match "Controllers: $ALICE_PRINCIPAL"
 
-    assert_command_fail dfx deploy --no-wallet
     assert_command_fail dfx deploy
-    assert_command dfx --identity alice deploy --no-wallet
+    assert_command dfx --identity alice deploy
 }
 
 @test "create accepts --controller <controller> named parameter, with controller by principal" {
     dfx_start
     dfx identity new alice
-    ALICE_PRINCIPAL=$(dfx --identity alice identity get-principal)
+    ALICE_WALLET=$(dfx --identity alice identity get-wallet)
 
-    assert_command dfx canister create --all --controller "${ALICE_PRINCIPAL}"
+    assert_command dfx canister create --all --controller "${ALICE_WALLET}"
```

Comment:
Again if we need to test passing a wallet ID, let's add a new, separate test, and leave this test (which tests passing a controller by principal).